pygmt/tests/test_grdgradient.py

```diff
@@ -0,0 +1,56 @@
+"""
+Tests for grdgradient.
+"""
+import os
+
+import numpy.testing as npt
+import pytest
+from pygmt import grdgradient, grdinfo
+from pygmt.datasets import load_earth_relief
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import GMTTempFile
+
+
+@pytest.fixture(scope="module", name="grid")
+def fixture_grid():
+    """
+    Load the grid data from the sample earth_relief file.
+    """
+    return load_earth_relief(resolution="01d", region=[-5, 5, -5, 5])
+
+
+def test_grdgradient_outgrid(grid):
+    """
+    Test the azimuth and direction parameters for grdgradient with a set
+    outgrid.
+    """
+    with GMTTempFile(suffix=".nc") as tmpfile:
+        result = grdgradient(grid=grid, outgrid=tmpfile.name, azimuth=10, direction="c")
+        assert result is None  # return value is None
+        assert os.path.exists(path=tmpfile.name)  # check that outgrid exists
+        result = (
+            grdinfo(grid=tmpfile.name, force_scan=0, per_column="n").strip().split()
+        )
+    npt.assert_allclose(float(result[4]), -0.0045060496)
+    npt.assert_allclose(float(result[5]), 0.0575332976)
```

Comment:
Check the median and mean for the NetCDF grid using `grdinfo`. Note that the values here differ from that of `xarray.DataArray` because GMT is calculating the "spherically weighted statistics" of the geographic grid (see https://docs.generic-mapping-tools.org/6.1/grdinfo.html#l) instead of a square pixel weighted one in the `xarray.DataArray`.

```suggestion
            grdinfo(grid=tmpfile.name, force_scan="a", per_column="n").strip().split()
        )
    npt.assert_allclose(float(result[4]), -0.0045060496)  # min
    npt.assert_allclose(float(result[5]), 0.0575332976)  # max
    # Check spherically weighted statistics below
    npt.assert_allclose(float(result[10]), 0.000384754501283)  # median
    npt.assert_allclose(float(result[12]), 0.00285958005568)  # mean
```